talpid-tunnel-config-client/proto/tunnel_config.proto

```diff
@@ -19,7 +7,69 @@ package tunnel_config;
 service PostQuantumSecure {
   // PskExchangeExperimentalV0 uses the common API defined by LibOQS.  See:
   // https://github.com/open-quantum-safe/liboqs
+  // This endpoint is deprecated in favor for `PskExchangeExperimentalV1`. Please use that instead.
   rpc PskExchangeExperimentalV0(PskRequestExperimentalV0) returns (PskResponseExperimentalV0) {}
+
+  // Allows deriving a preshared key (PSK) using one or multiple PQ-secure key-encapsulation
+  // mechanisms (KEM). The preshared key is added to WireGuard's preshared-key field in a new
+  // ephemeral peer (PQ-peer). This makes the tunnel resistant towards attacks using
+  // quantum computers.
+  //
+  // The VPN server associates the PQ-peer with the peer who performed the exchange. Any
+  // already existing PQ-peer for the normal peer is replaced. Each normal peer can have
+  // at most one PQ-peer.
+  //
+  // The PQ-peer is mutually exclusive to the normal peer. A handshake from the normal peer
+  // throws out the corresponding PQ-peer from WireGuard and vice versa.
+  //
+  // A new peer is negotiated for PQ to avoid a premature break of the tunnel used for negotiation.
+  // A tunnel would break prematurely if the preshared key is applied before the normal peer
+  // received the server's contribution to the KEM exchange. This cannot occur now because
+  // the client decides when to switch to the PQ-secure tunnel. This design also allows
+  // the client to switch back to using a non-PQ-secure tunnel at any point.
+  //
+  // The negotiated PQ-peer is ephemeral. The server gives no guarantees how long it will be
+  // valid and working. The client should negotiate a new PQ-peer every time it establishes a new
+  // tunnel to the server.
```

Comment:
This would be preferred. I was told that the server, by-design, has to keep the PQ-peer alive for as long as possible.

Reply:
What does "this" refer to here? What's the change you are suggesting? The main point I'm trying to convey to client implementors here is that the PQ-peer is not persisted on the server. That the ephemeral peer could be lost at any time.

Reply:
"This" refers to the behavior outlined in the paragraph above. `The client should negotiate ... every time` makes total sense but is in contrast to a previous paragraph where it's encouraged to switch back-and-forth between a client's normal peer and its PQ-peer.

Reply:
I don't think we encourage anything. We just say it's possible. And it's true, isn't it? We don't guarantee anything about how long the pq-peer will exist. But for as long as it does exist, the client can switch back and forth between the two peers freely. Not that any implementation we have will, but it's possible.